scene/gui/popup_menu.cpp

```diff
@@ -345,10 +345,14 @@ int PopupMenu::_get_mouse_over(const Point2 &p_over) const {
 }
 
 void PopupMenu::_activate_submenu(int p_over, bool p_by_keyboard) {
+	ERR_FAIL_INDEX_MSG(p_over, items.size(), vformat("Invalid submenu index %d in _activate_submenu.", p_over));
 	PopupMenu *submenu_popup = items[p_over].submenu;
-	if (submenu_popup->is_visible()) {
-		return; // Already visible.
-	}
+	ERR_FAIL_COND_MSG(submenu_popup->is_visible(), vformat("_activate_submenu should not be called on an open submenu - index: %d.", p_over));
+
+	submenu_popup->this_submenu_index = p_over; // `this_submenu_index` is always -1 for the parent popup, and always a positive int for every open submenu.
+	active_submenu_index = p_over; // A positive int for the parent popup if any submenu is open.
```

Comment:
These comment are already at the definition of these vars in the header, are they really needed?
Same with the other time `active_submenu_index` is set.